frontend/src/pages/AllChatsPage.jsx

```diff
@@ -23,16 +26,53 @@ const ChatInboxHeader = () => (
     </div>
 )
 
+// find the blocked accounts of the user
+function useFetchBlockedUsers(myUid) {
+    const [res] = useBackendFetchJson({
+        route: `/chat/block/user/${myUid}`,
+        deps: [myUid],
+        isValid: myUid,
+    })
+    const blockedUsers = res?.blocks || []
+    // create a blocked user dictionary
+    const reduceList = blockedUsers.reduce((accumulator, item) => {
+        if (item.blocked === true) {
+            accumulator[item.uid] = item.blocked
+        }
+        return accumulator
+    }, {})
+    return reduceList
+}
+
 export default function AllChatsPage() {
     const authUser = useCurrentAuthUser()
     const matches = useGetAllUserMatches(authUser?.uid)
+    const blockedUsers = useFetchBlockedUsers(authUser?.uid)
+    const [isOpen, setIsOpen] = useState(false)
+
+    const handleInboxOpening = () => {
+        setIsOpen((prev) => !prev)
+    }
+
+    // map to check if blocked user from dictionary is present in inbox
+    const matchesWithBlocks = matches.map((match) => ({
+        ...match,
+        hasBlockedUser: Object.keys(match.participants).some(
+            (participant) => participant in blockedUsers
+        ),
+    }))
+    // separate blocked and unblocked chats
+    const unblockedMatches = matchesWithBlocks.filter((m) => !m.hasBlockedUser)
+    const blockedMatches = matchesWithBlocks.filter((m) => m.hasBlockedUser)
```

Comment:
Nice! This is easier to follow. I know the code above is a little harder to read, but that's a consequence of how we have the match data structured. Good work!